_posts/2017-05-10-fastlane-travis-weekly-deploys.md

```diff
@@ -0,0 +1,249 @@
+
+```ruby
+# In case you need to update the signing profiles for this app
+lane :update_signing do
+  match(type: 'appstore')
+end
+
+# Used by CI, will not sneakily update (the CI only has read-only access to the repo anyway)
+lane :setup_signing do
+  match(type: 'appstore', readonly: true)
+end
+```
+
+## Creating the build and shipping it to Testflight
+
+This is handled by [fastlane gym][gym], again, I made a separate fastlane lane so it could be tested independently.
+
+```ruby
+# You can run `bundle exec fastlane build` to test the build process locally.
+lane :build do
+  gym workspace: 'emission/Example/Emission.xcworkspace',
+      configuration: 'Deploy',
+      scheme: 'Emission',
+      clean: true
+end
+```
+
+It uses a scheme for deploys, which prioritises using AppHub over a local React Native server. Gym handles a lot of CLI ugliness for us, and works well.
+
+Sending the app to Testflight involves a little bit of work:
+
+```ruby
+# Get the last 10 lines of the CHANGELOG for Testflight
+changelog = 'emission/CHANGELOG.md'
+upcoming_release_notes = File.read(changelog).lines[0...12].join("\n")
+
+# Log into iTunes connect, get the latest version of the app we shipped, and how many builds we've sent too
+Spaceship::Tunes.login(ENV['FASTLANE_USERNAME'], ENV['FASTLANE_PASSWORD'])
+app = Spaceship::Tunes::Application.find('net.artsy.Emission')
+latest_version = app.build_trains.keys.sort.last
+train = app.build_trains[latest_version]
+build_version = train.builds.count + 1
+```
+
+This lets the deploy process figure out what the latest release version is, and how many builds have shipped for that version. Then those can be used to set the build version and create a tag associated with it.
+
+```ruby
+# Ship to testflight
+pilot changelog: upcoming_release_notes
+
+# Tag releases
+add_git_tag tag: tag
+`git remote add http https://#{ENV["GITHUB_SUBMODULES_USER"]}@github.com/artsy/emission.git`
+`git push http #{tag}`
+```
+
+[fastlane pilot][] is send off the compiled build to Testflight, and then a tag noting the deploy is pushed to the main repo.
+
+## Notifications that it passed or succeeded.
+
+This was easy, I created a new slack inbound web-hook and added that as an environment variable. Then when a build passes we post a notification that there is a new version for everyone in Slack, if the lane fails then it will also post to slack. To ensure we keep on top of it, during development this was commented out.
+
+```ruby
+# If the weekly task fails, then ship a message
+error do |_, exception|
+   slack message: "Error Deploying Emission: #{exception}",
+         success: false,
+         payload: { Output: exception.error_info.to_s }
+end
+```
+
+That wraps up setting up the CI. Once you've confirmed everything has worked, you can add the scheduler inside Travis and expect to see a slack notification in a week.
```

Comment:
TIL that [Travis has cron jobs now](https://docs.travis-ci.com/user/cron-jobs/)! Diving into this could be a neat thing to shed some light on this feature.